testing/test_terminal.py

```diff
@@ -718,3 +719,63 @@ def pytest_terminal_summary(terminalreporter):
         *==== hello ====*
         world
     """)
+
+@pytest.mark.parametrize("exp_color, exp_line, stats_arg", [
+    # The method under test only cares about the length of each
+    # dict value, not the actual contents, so tuples of anything
+    # suffice
+
+    # Important statuses -- the highest priority of these always wins
+    ("red",    "1 failed",               {"failed":     (1,)}),
+    ("red",    "1 failed, 1 passed",     {"failed":     (1,), "passed": (1,)}),
+
+    ("red",    "1 error",                {"error":      (1,)}),
+    ("red",    "1 passed, 1 error",      {"error":      (1,), "passed": (1,)}),
+
+    ("red",    "1 xpassed",              {"xpassed":    (1,)}),
+    ("red",    "1 passed, 1 xpassed",    {"xpassed":    (1,), "passed": (1,)}),
+
+    # (a status that's not known to the code)
+    ("yellow", "1 weird",                {"weird":      (1,)}),
+    ("yellow", "1 passed, 1 weird",      {"weird":      (1,), "passed": (1,)}),
+
+    ("yellow", "1 warnings",             {"warnings":   (1,)}),
+    ("yellow", "1 passed, 1 warnings",   {"warnings":   (1,), "passed": (1,)}),
+
+    ("green",  "5 passed",               {"passed":     (1,2,3,4,5)}),
+
+
+    # Boring statuses -- if *every* test produced one of these, the status line
+    # is colored yellow to warn the user that the test run produced no useful
+    # information
+    ("yellow", "1 skipped",              {"skipped":    (1,)}),
+    ("green",  "1 passed, 1 skipped",    {"skipped":    (1,), "passed": (1,)}),
+
+    ("yellow", "1 deselected",           {"deselected": (1,)}),
+    ("green",  "1 passed, 1 deselected", {"deselected": (1,), "passed": (1,)}),
+
+    ("yellow", "1 xfailed",              {"xfailed":    (1,)}),
```

Comment:
From my understanding, `mark.xfail` is meant for flaky tests which are known to fail occasionally. If that's correct, I think `xfailed` and `xpassed` tests should be treated as "passed" for the purposes of this PR IMO.